src/Model/GContact.php

```diff
@@ -1056,21 +1061,23 @@ public static function discoverGsUsers()
 	}
 
 	/**
-	 * @return string
+	 * Returns a random, global contact of the current node
+	 *
+	 * @return string The profile URL
 	 * @throws Exception
 	 */
 	public static function getRandomUrl()
 	{
-		$r = q(
-			"SELECT `url` FROM `gcontact` WHERE `network` = '%s'
-					AND `last_contact` >= `last_failure`
-					AND `updated` > UTC_TIMESTAMP - INTERVAL 1 MONTH
-				ORDER BY rand() LIMIT 1",
-			DBA::escape(Protocol::DFRN)
-		);
+		$r = DBA::selectFirst('gcontact', ['url'], [
+			'`network` = ? 
+			AND `last_contact` >= `last_failure`  
+			AND	`updated` > ?',
+			Protocol::DFRN,
+			DateTimeFormat::utc('now - 1 month'),
+		], ['order' => ['RAND()']]);
 
 		if (DBA::isResult($r)) {
-			return dirname($r[0]['url']);
```

Comment:
Why did you remove the `dirname()` call?

Reply:
Because the `dirname` removed the last part of the url, so I got redirected to `friendica.local/profile` all the time. After removing it, I was correctly redirected to `friendica.local/profile/friendica1`